sdk/python/tests/test_historical_retrieval.py

```diff
@@ -560,3 +608,79 @@ def test_historical_features_from_bigquery_sources(
             .reset_index(drop=True),
             check_dtype=False,
         )
+
+
+@pytest.mark.integration
+def test_feature_name_collision_on_historical_retrieval_from_parquet_sources():
+    start_date = datetime.now().replace(microsecond=0, second=0, minute=0)
+    (customer_entities, driver_entities, end_date, _, start_date,) = generate_entities(
+        start_date, True
+    )
+
+    with TemporaryDirectory() as temp_dir:
+        driver_df = driver_data.create_driver_hourly_stats_df(
+            driver_entities, start_date, end_date
+        )
+        driver_source = stage_driver_hourly_stats_parquet_source(temp_dir, driver_df)
+        driver_fv = create_driver_hourly_stats_feature_view(driver_source)
+        customer_df = driver_data.create_customer_daily_profile_df(
+            customer_entities, start_date, end_date
+        )
+        customer_source = stage_customer_daily_profile_parquet_source(
+            temp_dir, customer_df
+        )
+        customer_fv = create_customer_daily_profile_feature_view(customer_source)
+
+        # _group_refs is the function that checks for colliding feature names
+        with pytest.raises(FeatureNameCollisionError):
+            _group_refs(
+                feature_refs=[
+                    "driver_stats:conv_rate",
+                    "driver_stats:avg_daily_trips",
+                    "customer_profile:current_balance",
+                    "customer_profile:avg_passenger_count",
+                    "customer_profile:lifetime_trip_count",
+                    "customer_profile:avg_daily_trips",
+                ],
+                all_feature_views=[driver_fv, customer_fv],
+                full_feature_names=False,
+            )
+
+
+def test_feature_name_collision_on_historical_retrieval_from_bigquery_sources():
+    bigquery_dataset = (
+        f"test_hist_retrieval_{int(time.time_ns())}_{random.randint(1000, 9999)}"
+    )
+    gcp_project = "project_name"
+
+    # Driver Feature View
+    driver_table_id = f"{gcp_project}.{bigquery_dataset}.driver_hourly"
+    driver_source = BigQuerySource(
+        table_ref=driver_table_id,
+        event_timestamp_column="datetime",
+        created_timestamp_column="created",
+    )
+    driver_fv = create_driver_hourly_stats_feature_view(driver_source)
+
+    customer_table_id = f"{gcp_project}.{bigquery_dataset}.customer_profile"
+    customer_source = BigQuerySource(
+        table_ref=customer_table_id,
+        event_timestamp_column="datetime",
+        created_timestamp_column="",
+    )
+    customer_fv = create_customer_daily_profile_feature_view(customer_source)
+
+    # _group_refs is the function that checks for colliding feature names
+    with pytest.raises(FeatureNameCollisionError):
+        _group_refs(
```

Comment:
Do we need to write to BigQuery in order to test a function? This seems like a unit test.

Reply:
I don't think I write to BigQuery here, but I did notice that I do write to a temporary directory in the test above the one you highlighted (using parquet sources) which I will change. Do let me know if I'm missing something.